selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/CSharpFileEditingTest.java

```diff
@@ -56,69 +56,80 @@
+    if (testWorkspaceServiceClient.getStatus(workspace.getId()).equals(WorkspaceStatus.RUNNING)) {
+      editor.closeAllTabs();
+      testWorkspaceServiceClient.stop(workspace.getName(), workspace.getOwner().getName());
+      ide.open(workspace);
+      ide.waitOpenedWorkspaceIsReadyToUse();
+      restoreLanguageServer();
+      projectExplorer.quickRevealToItemWithJavaScript(PROJECT_NAME + "/Program.cs");
+    }
   }
 
   @Test
-  public void checkLaunchingCodeserver() {
-    projectExplorer.waitProjectExplorer();
-    menu.runCommand(
-        TestMenuCommandsConstants.Workspace.WORKSPACE,
-        TestMenuCommandsConstants.Workspace.CREATE_PROJECT);
-    wizard.selectSample(Wizard.SamplesName.ASP_DOT_NET_WEB_SIMPLE);
-    wizard.typeProjectNameOnWizard(PROJECT_NAME);
-    wizard.clickCreateButton();
-    wizard.waitCloseProjectConfigForm();
-    projectExplorer.openItemByPath(PROJECT_NAME);
-    projectExplorer.waitItem(PROJECT_NAME + "/Program.cs", 240);
+  public void checkCodeEditing() {
     projectExplorer.openItemByPath(PROJECT_NAME + "/Program.cs");
-    loader.waitOnClosed();
-    checkLanguageServerInitStateAndLaunch();
+    checkCodeValidation();
+  }
+
+  @Test(priority = 1)
+  public void checkInitializingAfterFirstStarting() {
+    projectExplorer.openItemByPath(PROJECT_NAME + "/Program.cs");
+    try {
+      editor.waitMarkerInPosition(INFO, 1);
+      editor.waitMarkerInPosition(INFO, 2);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("https://github.com/eclipse/che/issues/10151");
+    }
+  }
+
+  public void checkCodeValidation() {
     editor.goToCursorPositionVisible(24, 12);
     for (int i = 0; i < 9; i++) {
       editor.typeTextIntoEditor(Keys.BACK_SPACE.toString());
     }
     editor.waitMarkerInPosition(ERROR, 23);
     editor.waitMarkerInPosition(ERROR, 21);
+    checkAutocompletion();
+  }
+
+  private void checkAutocompletion() {
     editor.goToCursorPositionVisible(23, 49);
     editor.typeTextIntoEditor(".");
     editor.launchAutocomplete();
-    editor.enterAutocompleteProposal("Build() ");
-    editor.typeTextIntoEditor(";");
+    editor.enterAutocompleteProposal("Build ");
+    editor.typeTextIntoEditor("();");
     editor.waitAllMarkersInvisibility(ERROR);
   }
 
-  private void checkLanguageServerInitStateAndLaunch() {
-    if (isLanguageServerInitFailed()) {
-      reInitLanguageServer();
-    }
+  private void createDotNetAppFromWizard() {
+    projectExplorer.waitProjectExplorer();
+    menu.runCommand(
+        TestMenuCommandsConstants.Workspace.WORKSPACE,
+        TestMenuCommandsConstants.Workspace.CREATE_PROJECT);
+    wizard.selectSample(Wizard.SamplesName.ASP_DOT_NET_WEB_SIMPLE);
+    wizard.typeProjectNameOnWizard(PROJECT_NAME);
+    wizard.clickCreateButton();
+    wizard.waitCloseProjectConfigForm();
   }
 
-  private void reInitLanguageServer() {
+  private void restoreLanguageServer() {
```

Comment:
We are not restoring language server directly here, aren't we?
We are doing something which lead to restoring language server. It's kind of magic which have to be commented clearly, and method name should reflect the magic explicitely.